option_test.go

```diff
@@ -1073,3 +1076,88 @@ func TestOptionStripTrailingSlash(t *testing.T) {
 		require.Equal(t, "/test", route.Path)
 	})
 }
+
+// kvSerde has custom serialization and deserialization logic for key-value pairs
+// in the format "key1=value1;key2=value2;..."
+type kvSerde struct{}
+
+// Serialize custom serialization logic
+func (s kvSerde) Serialize(v any) ([]byte, error) {
+	v2, ok := v.(map[string]string)
+	if !ok {
+		return nil, errors.New("invalid type; kvSerde expected map[string]string")
+	}
+	parts := make([]string, 0, len(v2))
+	for k, v := range v2 {
+		parts = append(parts, fmt.Sprintf("%s=%s", k, v))
+	}
+	slices.Sort(parts)
+	joined := strings.Join(parts, ";")
+	return []byte(joined), nil
+}
+
+// Deserialize custom deserialization logic
+func (s kvSerde) Deserialize(data []byte) (any, error) {
+	v := map[string]string{}
+	pairs := strings.Split(string(data), ";")
+
+	for _, pair := range pairs {
+		kv := strings.Split(pair, "=")
+		if len(kv) != 2 {
+			return nil, errors.New("invalid format; kvSerde expected key=value pairs separated by ;")
+		}
+		v[kv[0]] = kv[1]
+	}
+	return v, nil
+}
+
+func TestWithCustomSerde(t *testing.T) {
+	controller := func(c fuego.ContextWithBody[map[string]string]) (map[string]string, error) {
+		body, err := c.Body()
+		if err != nil {
+			return nil, err
+		}
+		body["foo"] = "bar"
+		return body, nil
+	}
+
+	t.Run("Custom serde is used", func(t *testing.T) {
+		s := fuego.NewServer()
+		fuego.Post(s, "/keyvalue", controller,
+			option.WithContentTypeSerde("application/vnd.keyvalue", kvSerde{}),
+		)
+
+		// In kv; out kv
+		r := httptest.NewRequest("POST", "/keyvalue", strings.NewReader("key1=hello;key2=2"))
+		r.Header.Set("Accept", "application/vnd.keyvalue")
+		r.Header.Set("Content-Type", "application/vnd.keyvalue")
+		w := httptest.NewRecorder()
+
+		s.Mux.ServeHTTP(w, r)
+
+		require.Equal(t, 200, w.Code)
+		require.Equal(t, "foo=bar;key1=hello;key2=2", w.Body.String())
+
+		// In kv; out json
+		r = httptest.NewRequest("POST", "/keyvalue", strings.NewReader("key1=hello;key2=2"))
+		r.Header.Set("Accept", "application/json")
+		r.Header.Set("Content-Type", "application/vnd.keyvalue")
+		w = httptest.NewRecorder()
+
+		s.Mux.ServeHTTP(w, r)
+
+		require.Equal(t, 200, w.Code)
+		require.JSONEq(t, `{"foo":"bar","key1":"hello","key2":"2"}`, w.Body.String())
+
+		// In json; out kv
```

Comment:
Instead of separating with comments lets use table driven tests. Your comments for the tittle would work great there